### DISCUSSION TITLE
How to encode abi for tuple parameters?

### DISCUSSION THREAD

**Discussion by 0xguystar007** [wevm/viem#659] [Question]
const encodedParams = encodeAbiParameters(parseAbiParameters("address, tuple(bytes3,address,uint256,bytes)[]"), [
        myAddress,
        data,
      ]);

I was trying to get encoded abi. but it says `InvalidParameterError: Invalid ABI parameter`.
How to figure it out? Please help me.

**Comment by jxom** [comment]
`tuple` isn't a valid human-readable keyword. you can remove it. `"address, (bytes3,address,uint256,bytes)[]"`

**Reply by 0xguystar007** [reply]
```
   const data = [{
      standard: "0x0000" as `0x${string}`,
      addr: "0x0987654321098765432109876543210987654321" as `0x${string}`,
      id: 0n,
      data: "0x0000" as `0x${string}`,
    }];
```

Here is the value of data const.
Please let me know how to get encoded abi parameter from it. @jxom
